src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala

```diff
@@ -155,94 +159,30 @@ object GermlineAssemblyCaller {
-          AlleleEvidence(
-            likelihood = 1,
-            readDepth = depth,
-            alleleReadDepth = depth,
-            forwardDepth = depth,
-            alleleForwardDepth = depth,
-            meanMappingQuality = mean(mappingQualities),
-            medianMappingQuality = median(mappingQualities),
-            meanBaseQuality = mean(baseQualities),
-            medianBaseQuality = median(baseQualities),
-            medianMismatchesPerRead = 0
-          )
-        )
-      }
-
-      val variants =
-        pathAndAlignments.flatMap(kv => {
-          val path = kv._1
-          val alignment = kv._2
-
-          var referenceIndex = alignment.refStartIdx
-          var pathIndex = 0
-
-          // Find the alignment sequences using the CIGAR
-          val cigarElements = alignment.toCigar.getCigarElements
-
-          cigarElements.flatMap(cigarElement => {
-            val cigarOperator = cigarElement.getOperator
-            val referenceLength = CigarUtils.getReferenceLength(cigarElement)
-            val pathLength = CigarUtils.getReadLength(cigarElement)
-
-            // Yield a resulting variant when there is a mismatch, insertion or deletion
-            val possibleVariant = cigarOperator match {
-              case CigarOperator.X =>
-                val referenceAllele = currentReference.slice(referenceIndex, referenceIndex + referenceLength)
-                val alternateAllele = path.slice(pathIndex, pathIndex + pathLength)
-                Some(buildVariant(referenceIndex, referenceAllele, alternateAllele.toArray))
-              case (CigarOperator.I | CigarOperator.D) if referenceIndex != 0 =>
-                // For insertions and deletions, report the variant with the last reference base attached
-                val referenceAllele = currentReference.slice(referenceIndex - 1, referenceIndex + referenceLength)
-                val alternateAllele = path.slice(pathIndex - 1, pathIndex + pathLength)
-                Some(buildVariant(referenceIndex - 1, referenceAllele, alternateAllele.toArray))
-              case _ => None
-            }
-
-            referenceIndex += referenceLength
-            pathIndex += pathLength
-
-            possibleVariant
+        .map(DeBruijnGraph.mergeOverlappingSequences(_, kmerSize))
+        .toVector
+
+      // Score up to the maximum number of paths
+      if (paths.size == 0) {
+        log.warn(s"In window ${referenceContig}:${referenceStart}-$referenceEnd " +
+          s"assembly failed")
+        List.empty
+      } else if (paths.size <= expectedPloidy) {
+        paths
+      } else if (paths.size <= maxPathsToScore) {
+        val pathScores = DenseVector.zeros[Int](paths.size)
+        reads.foreach(
+          read => {
+            pathScores :+= DenseVector(paths.map(path => -AffineGapPenaltyAlignment.align(read.sequence, path).alignmentScore): _*)
           })
 
-        })
+        argtopk(pathScores, expectedPloidy).map(paths(_))
```

Comment:
I think in this implementation it won't matter, so I can simplify it as you stated. This originally developed from building the full matrix of path/read scores ... but I will think about this a bit